### PR TITLE
Complete rewrite

### PR DESCRIPTION
**Pull request by shinokada**
- Private repository support (-p/--private)
- Custom commit messages (-m/--message)
- Custom branch names (-b/--branch)
- Repository descriptions (-desc/--description)
- Dry-run mode (--dry-run)
- Quiet mode (-q/--quiet)
- Better error handling with automatic cleanup
- Full support for existing directories
- Fixed GitHub auth check bug
- XDG-compliant config location

uninstall.sh - Enhanced to handle both old and new config locations docs/README.md - Updated with all new features
README.md (root) - Comprehensive project documentation

Closes # 

## ✅ Checks

- [ ] My pull request adheres to the code style of this project
- [ ] My code requires changes to the documentation
- [ ] I have updated the documentation as required
- [ ] All the tests have passed

## Summary by CodeRabbit

* **New Features**
  * Dry-run and quiet non‑interactive modes; private repos, custom branch/message/description; XDG config support and migration; improved existing‑directory handling

* **Documentation**
  * Expanded README, changelog (v0.4.0), migration guide, examples, testing and cross‑platform guides, quick reference, and many HOWTOs

* **Tests**
  * New test suites and runners, static analysis integration, CI workflow, and permission/verification helpers

* **Bug Fixes**
  * Absolute path handling, non‑interactive prompt blocking, cloning/checkout edge cases

* **Chores**
  * Makefile, install/uninstall improvements, repository attributes and cleanup docs

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: updates/CROSS_PLATFORM.md [hunk 1]
```diff
@@ -0,0 +1,453 @@
...
+### 1. File System Paths ✅
+
+**Issue:** Path handling differs between systems
+**Solution:** Uses POSIX-compliant path operations
+
+```bash
+# Works on both platforms
+dir=$(pwd)
+dir="$(pwd)/$dir"
+```
```
[the remark this answers is not in the document]
Comment (shinokada) [reply updates/CROSS_PLATFORM.md]: @coderabbitat
FYI: Regarding above suggestions, the following is what I did and I will push it later

## About GitHub Actions Suggestions

Skip them - They don't apply to our workflow
We run shellcheck directly (better than action wrapper)
We don't use artifact uploads or Trivy
GitHub Security bot will handle CodeQL

## About fix-permissions.sh Warning

Keep it - It's a useful helper tool
Just make it executable: chmod +x fix-permissions.sh
Add note in README if you want
See updates/ABOUT_FIX_PERMISSIONS.md for details

## About GitHub Advanced Security Bot

ACCEPT IT! - It's free security scanning
Catches vulnerabilities automatically
May flag false positives (easy to dismiss)
No downside for public repos

File: updates/CI_SETUP.md [hunk 2]
```diff
@@ -0,0 +1,352 @@
...
+✓ Checkout code
+✓ Install dependencies  
+✓ Verify dependencies
+✓ Fix script permissions
+✓ Run ShellCheck
+✓ Run unit tests
+✓ Test summary
+
+All checks have passed
+```
```
[the remark this answers is not in the document]
Comment (shinokada) [reply updates/CI_SETUP.md]: All markdown files except README are ignored.

File: gitstart [hunk 3]
```diff
@@ -1,252 +1,418 @@
...
+gh auth status &>/dev/null || error "Run 'gh auth login' first"
 
-############# Main body ############
-# check if you are logged in github
-if [[ $(gh auth status) -eq 1 ]]; then
-    # not logged-in
-    echo ">>> You must logged in. Use 'gh auth login'"
-    exit 1
-fi
+[[ -n "${dir}" ]] || error "Directory is required (-d)"
```
[the remark this answers is not in the document]
Comment (shinokada) [reply gitstart]: fixed.

File: verify-changes.sh [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+echo "3. .github/workflows/tests.yml - Shell syntax"
+echo "   Line 54: Using [[ ]] with quotes"
+grep -A 1 'job.status' .github/workflows/tests.yml || echo "   ❌ Not found!"
+echo ""
+
+echo "4. tests/run-tests.sh - Simplified conditional"
+echo "   Checking CI_ENV conditional..."
+grep -A 2 'CI_ENV.*true' tests/run-tests.sh | head -10
+echo ""
```
[the remark this answers is not in the document]
Comment (shinokada) [reply verify-changes.sh]: fixed.

File: gitstart [hunk 5]
```diff
@@ -1,252 +1,423 @@
...
+        MIT)
+            license_url="mit"
+            break
+            ;;
+        Apache-2.0)
+            license_url="apache-2.0"
+            break
+            ;;
+        GPLv3)
+            license_url="lgpl-3.0"
```
[the remark this answers is not in the document]
Comment (shinokada) [reply gitstart]: fixed.

File: tests/test-path-handling.sh [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+fi
+echo ""
+
+echo "Test 4: Empty commit message"
+echo "---------------------------"
+if "${GITSTART}" -d test-repo -m "" --dry-run 2>&1 | grep -q "Commit message cannot be empty"; then
+    echo "✓ Empty commit message rejected"
+else
+    echo "✗ Empty commit message not caught"
+    exit 1
```
[the remark this answers is not in the document]
Comment (shinokada) [reply tests/test-path-handling.sh]: fixed.

File: tests/test-dry-run-simple.sh [hunk 7]
```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env bash
+
+# Quick test for dry-run mode in non-interactive CI environment
+
+set -euo pipefail
+
+SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+GITSTART="${SCRIPT_DIR}/gitstart"
+
```
[the remark this answers is not in the document]
Comment (shinokada) [reply tests/test-dry-run-simple.sh]: fixed.

File: tests/test-validation.sh [hunk 8]
```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env bash
+
+# Test validation improvements
+
+set -euo pipefail
+
+SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+GITSTART="${SCRIPT_DIR}/gitstart"
```
[the remark this answers is not in the document]
Comment (shinokada) [reply tests/test-validation.sh]: fixed.
